webapp/src/components/common/fieldEditors/NewSelectFE.tsx

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2025, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import reactHookFormSupport, { type ReactHookFormSupportProps } from "@/hoc/reactHookFormSupport";
+import type { SvgIconComponent } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  Chip,
+  MenuItem,
+  TextField,
+  Tooltip,
+  type TextFieldProps,
+} from "@mui/material";
+import startCase from "lodash/startCase";
+import * as RA from "ramda-adjunct";
+import { useMemo } from "react";
+import type { FieldPath, FieldValues } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+
+type Value = string | number;
+
+interface Option<T extends Value> {
+  label: string;
+  value: T;
+  icon?: SvgIconComponent;
+  tooltip?: string;
+}
+
+type Options<T extends Value> = Array<T | Option<T>>;
+
+interface SingleSelect {
+  multiple?: false;
+  value?: Value;
+  defaultValue?: Value;
+  onSelectAllOptions?: never;
+  onDeselectAllOptions?: never;
+}
+
+interface MultipleSelect<OptionValue extends Value = Value> {
+  multiple: true;
+  // `value` or `defaultValue` may differ from `options` at initialization.
+  // Reason why `Value` is used instead of `OptionValue`.
+  value?: Value[];
+  defaultValue?: Value[];
+  onSelectAllOptions?: (optionValues: OptionValue[]) => void;
+  onDeselectAllOptions?: VoidFunction;
+}
+
+type NewSelectFEProps<OptionValue extends Value = Value> = {
+  options?: Options<OptionValue>;
+  emptyValue?: boolean;
+  startCaseLabel?: boolean;
+  renderValueAs?: "text" | "chip";
+} & (SingleSelect | MultipleSelect<OptionValue>) &
+  Omit<TextFieldProps, "select" | "type" | "value" | "defaultValue">;
+
+function NewSelectFE<OptionValue extends Value = Value>({
```

Comment:
- you can rename it `SelectFEv2` to clearly indicate it's version 2 of the SelectFE component and make the migration intent obvious 

- if you do not plan to make the migration and remove the `SelectFE` soon, you can create a Jira technical story to track it


- you can also add documentation on the usage of the API, since you decided to mix the 2 use-cases (single vs multiple) with examples of usage, and add few basic unit tests on important logic

Reply:
I will make a new PR for the migration soon. I added a deprecated tag to the old Select.